lib/coinbase/smart_contract.rb

```diff
@@ -23,63 +23,64 @@ def self.list_events(
     )
       Coinbase::Pagination.enumerate(
         lambda { |page|
-          list_events_page(network_id, protocol_name, contract_address, contract_name, event_name, from_block_height,
-                           to_block_height, page)
+          list_events_page(
+            network_id,
+            protocol_name,
+            contract_address,
+            contract_name,
+            event_name,
+            from_block_height,
+            to_block_height,
+            page
+          )
         }
       ) do |contract_event|
         Coinbase::ContractEvent.new(contract_event)
       end
     end
 
-    # Returns a new SmartContract object.
-    # @param model [Coinbase::Client::SmartContract] The underlying SmartContract object
-    def initialize(model)
-      @model = model
-    end
-
-    # Returns the network ID of the SmartContract.
-    # @return [String] The network ID
-    def network_id
-      Coinbase.to_sym(@model.network_id)
-    end
-
-    # Returns the protocol name of the SmartContract.
-    # @return [String] The protocol name
-    def protocol_name
-      @model.protocol_name
-    end
-
-    # Returns the contract name of the SmartContract.
-    # @return [String] The contract name
-    def contract_name
-      @model.contract_name
-    end
-
-    # Returns the address of the SmartContract.
-    # @return [String] The contract address
-    def address
-      @model.address
-    end
-
-    # Returns a string representation of the SmartContract.
-    # @return [String] a string representation of the SmartContract
-    def to_s
-      "Coinbase::SmartContract{
-        network_id: '#{network_id}',
-        protocol_name: '#{protocol_name}',
-        contract_name: '#{contract_name}',
-        address: '#{address}'}"
-    end
+    # Creates a new ERC20 token contract, that can subsequently be deployed to
+    # the blockchain.
+    # @param address_id [String] The address ID of deployer
+    # @param wallet_id [String] The wallet ID of the deployer
+    # @param name [String] The name of the token
+    # @param symbol [String] The symbol of the token
+    # @param total_supply [String] The total supply of the token, denominate in whole units.
+    # @return [SmartContract] The new ERC20 Token SmartContract object
+    def self.create_token_contract(
```

Comment:
in the Node SDK, i put this same code in `wallet_address.ts`. Should either this change or the Node SDK change or is ok that the code lives in different places?

Reply:
Good question, in ruby this always lives with the model.

In NodeJS, it has lived on the other model (which I think isn't ideal here, but is a bigger change).